google/cloud/aiplatform/jobs.py

```diff
@@ -1913,3 +1918,507 @@ def run(
+    def _parse_configs(
+        cls,
+        objective_configs: Union[
+            model_monitoring.EndpointObjectiveConfig,
+            Dict[str, model_monitoring.EndpointObjectiveConfig],
+        ],
+        endpoint: "aiplatform.Endpoint",
+        deployed_model_ids: Optional[List[str]] = None,
+    ) -> List[
+        gca_model_deployment_monitoring_job_compat.ModelDeploymentMonitoringObjectiveConfig
+    ]:
+        """Helper function for matching objective configs with their corresponding models.
+
+        Args:
+            objective_configs (Union[model_monitoring.objective.EndpointObjectiveConfig,
+                Dict[str, model_monitoring.objective.EndpointObjectiveConfig]):
+                Required. A single config if it applies to all models, or a dictionary of
+                model_id: model_monitoring.objective.EndpointObjectiveConfig if
+                different model IDs have different configs.
+            endpoint (aiplatform.Endpoint):
+                Required. A valid instance of aiplatforn.Endpoint to launch the MDM job on.
+            deployed_model_ids (Optional[List[str]]):
+                Optional. A list of deployed model IDs to apply the objective config to.
+                Note that a model will have a deployed_model_id that is different from the
+                uploaded model ID, and IDs in this list should consist of deployed model IDs
+                on the same endpoint passed in the argument. If `objective_configs` is a dictionary,
+                then this parameter is ignored. If `objective_configs` is an instance of
+                `model_monitoring.EndpointObjectiveConfig` and `deployed_model_ids` is a non-empty
+                list of valid IDs, then the same objective config will apply to all models in this list.
+
+        Returns:
+            A List of ModelDeploymentMonitoringObjectiveConfig objects.
+
+        Raises:
+            ValueError, when the model IDs given are invalid.
+            RuntimeError, when XAI is enabled on a model that doesn't have XAI parameters configured.
+        """
+        all_models = []
+        xai_enabled = []
+        for model in endpoint.list_models():
+            all_models.append(model.id)
+            if str(model.explanation_spec.parameters) != "":
+                xai_enabled.append(model.id)
+
+        all_configs = []
+
+        ## when same objective config is applied to SOME or ALL models
+        if deployed_model_ids is not None and not all(
+            model in all_models for model in deployed_model_ids
+        ):
+            error_string = (
+                "Invalid model ID. The model ID must be one of ["
+                + ",".join(all_models)
+                + "]. Note that deployed model IDs are different from the uploaded model's ID"
+            )
+            raise ValueError(error_string)
+
+        if isinstance(objective_configs, model_monitoring.EndpointObjectiveConfig):
+            for model in all_models:
```

Comment:
When `deployed_model_ids` is provided by the called, should the objective config be applied only to those deployed models instead of `all_models`?

Reply:
I added a block above this one to replace `all_models` with `deployed_model_ids` if `deployed_model_ids` is supplied and all the model IDs are valid.
```
        if deployed_model_ids is not None:
            if not all(model in all_models for model in deployed_model_ids):
                error_string = (
                    "Invalid model ID. The model ID must be one of ["
                    + ",".join(all_models)
                    + "]. Note that deployed model IDs are different from the uploaded model's ID"
                )
                raise ValueError(error_string)
            else:
                all_models = deployed_model_ids
```

Reply:
Thanks - the `_parse_configs` method needs to handle several different combinations of arguments.  Do we have some tests for it (or the public method that calls it) to ensure that the behavior is as expected?